src/client/txrx_client.cpp

```diff
@@ -503,14 +500,21 @@ void* RadioTXRX::loop_tx_rx_argos_sync(int tid)
     int rx_offset(0);
     size_t cursor(0);
     std::stringstream sout;
+
+    // Keep receiving one frame of data until a beacon is found
+    // Perform initial beacon detection every 10 frames
     while (c->running && sync_index < 0) {
-        int r = radio->radioRx(
-            radio_id, frm_rx_buff.data(), frm_num_samps, rxTime);
+        int r;
+        for (int find_beacon_retry = 0; find_beacon_retry < 10;
```

Comment:
It is better to replace `10` here with a global variable. You can add it to src/common/Symbols.hpp, e.g., `static constexpr size_t kBeaconDetectInterval = 10`, and also add the corresponding comment there.

Reply:
fixed!